### ISSUE TITLE
Feature query?  Outlook365

### ISSUE THREAD

**Issue by PeterS242** [lefcha/imapfilter#221]
Our org switched to using Outlook365 within the last year and I've not been able to use ImapFilter since for obvious reasons.  I was just wondering, are there any plans to support Outlook365?  It is pretty messy and requires quite a bit of underpinnings, so I'd understand.  It was a challenge just getting my Gnome Evolution app to access it, however it would be amazing if ImapFilter might support it at some point.

**Comment by lefcha** [comment]
Does this support access through an IMAP server? Something is mentioned in the [POP and IMAP email settings for Outlook](https://support.microsoft.com/en-us/office/pop-and-imap-email-settings-for-outlook-8361e398-8af4-4e97-b147-6c6c4ac95353) page, about a `outlook.office365.com` IMAP server. Can this be what you need to use?

If access through an IMAP server is not supported, then unfortunately it's out of scope of imapfilter...

**Comment by PeterS242** [comment]
> Does this support access through an IMAP server? Something is mentioned in the [POP and IMAP email settings for Outlook](https://support.microsoft.com/en-us/office/pop-and-imap-email-settings-for-outlook-8361e398-8af4-4e97-b147-6c6c4ac95353) page, about a `outlook.office365.com` IMAP server. Can this be what you need to use?
> 
> If access through an IMAP server is not supported, then unfortunately it's out of scope of imapfilter...

Well in my case IMAP is not allowed for accessing these Outlook mailboxes as MFA is not supported by IMAP.  And as far as being out-of-scope, I can completely understand your argument, however OATH appears to be supported now by ImapFilter which isn't IMAP and seems to be in-scope.  Also I'm still wanting to access and transport email to and from internal on-site IMAP/IMAPS servers so I don't know what other tool I would use to perform this other than ImapFilter.  I'm open to suggestions on that.  Honestly I wouldn't mind using a IMAP-to-O365 proxy of some kind, that would probably get me access to O365 using ImapFilter but I know of no such tools.

**Comment by PeterS242** [comment]
I apologize, the OAuth support is just for authentication within an IMAP conversation, so I think your argument applies.  Any suggestions are welcome.

**Comment by lefcha** [comment]
Yes, the OAuth2 support was part of the IMAP `AUTHENTICATE`, and specifically the `SASL XOAUTH2` mechanism.  It was pretty easy to support, just pass a string instead of a username/password, so I added support for it.

Now regarding this and MFA, it seems that mail providers are removing IMAP, and others are providing it as a legacy way to access email, but who knows for how long...

Not sure how I can help with Outlook365 support though. I mean I have no idea what protocols it uses, whether they are Internet standards, or how does Evolution implement its support...

**Comment by PeterS242** [comment]
@lefcha thank you very much for your consideration.  I find ImapFilter to be insanely useful and it's pretty difficult for me living without it.  It does seem that IMAP appears to be going the way of the Dodo, are there any plans to potentially move away from _only_ IMAP/IMAPS support and maybe have a more pluggable access available?  I suppose I'm asking if there are plans for the future?

Perhaps I need to take this up in a mailing list discussion, and if so I'm happy to switch to that and close this "Issue".  Or perhaps what I'm asking is already being discussed.

**Comment by lefcha** [comment]
Which mailing list? Please keep me updated, as I have lost touch with the IMAP and in general mailing protocols situation...

**Comment by robertek** [comment]
My org also switched to outlook recently and I continue using the imapfilter using the legacy username and pass now. But this seem to be only till the end of the 2021. After that there will be only oauth2.

**Comment by PeterS242** [comment]
> My org also switched to outlook recently and I continue using the imapfilter using the legacy username and pass now. But this seem to be only till the end of the 2021. After that there will be only oauth2.

@robertek FWIW, the way I get around it today is I run a Fedora Rawhide VM using Evolution, and one of our email administrators enabled a feature on our O365 Tenant that allows me to connect using Evolution 2FA.  From time to time I have to re-authenticate it which involves a pseudo web browser session within Evolution to complete.  I still would love to have IMAPFILTER usable again someday in whatever form that would be.

**Comment by dammmn** [comment]
For everyone using O365: in my case, org disabled imap and switched fully into m$ auth method (whatever that is). [Davmail](http://davmail.sourceforge.net/) was a way for me to keep my settings and get my email from that m$ monster. Apologies for off-topic.

**Comment by websafe** [comment]
@dammmn It's not off-topic - davmail is some way around but not really the best solution.
